zerver/actions/message_send.py

```diff
@@ -499,8 +499,8 @@ def build_message_send_dict(
     testing code paths call it directly.
     """
     if realm is None:
-        realm = message.sender.realm
-    assert realm == message.realm
+        realm = message.realm
+    assert realm.id == message.realm_id
```

Comment:
I think this is always called with an actual `Realm` object; should we just make `realm` a required non-optional parameter for this function? Or remove the parameter from this function and just unconditionally get it from `message.realm`? The latter might be cleaner.

I think cleaning this up is basically a follow-up refactor to the addition of the `message.realm` field, and while it's correct for `check_message` to default to `sender.realm` if needed, I think this function should be able to just expect to get a `Realm` object the normal way.

Reply:
Hmm yeah that would make sense, I'll see if this can be cleaned up

Reply:
@timabbott Updated the PR - 72950281943fdc662adf6515a390893a9b857b46 already had the optimization, so I'm just doing the arg cleanup dicussed here:

> Or remove the parameter from this function and just unconditionally get it from message.realm? The latter might be cleaner.

The arg seems redundant, since we're asserting it to end up equal to `message.realm` - so I pushed a commit to remove it